### PR TITLE
Remove method Chaining from videojs

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
* Remove method chaining from video.js
* Returning a Promise from play should happen in a separate PR

## Specific Changes proposed
List of methods that were changed:
* src/js/component.js
  * ready
  * addClass
  * removeClass
  * toggleClass
  * show
  * hide
  * lockShowing
  * unlockShowing
  * setAttribute
  * removeAttribute
  * dimension - (Only when setting)
  * dimensions
  * on
  * off
  * one
  * trigger
  * height - (only when setting)
  * width - (only when setting)
* src/js/modal-dialog.js
  * close
  * open
  * empty
  * fill
  * fillWith
* src/js/slider/slider.js
  * vertical - (only when setting)
* src/js/tech/tech.js
  * withSourceHandlers/setSource
* src/js/clickable-component.js
  * controlText - getter
  * enable
  * disable
* src/js/menu/menu-button.js
  * enable
  * disable
* src/js/player.js
  * pause
  * play
  * scrubbing - (only when setting)
  * currentTime - (only when setting)
  * duration - (only when setting)
  * volume - (only when setting)
  * hasStarted - (only when setting)
  * muted - (only when setting)
  * isFullscreen - (only when setting)	
  * requestFullscreen
  * exitFullscreen
  * src - (only when setting)
  * load
  * reset
  * preload - (only when setting)
  * autoplay - (only when setting)
  * loop - (only when setting)
  * poster - (only when setting)
  * controls - (only when setting)
  * usingNativeControls - (only when setting)
  * playbackRate - (only when setting)
  * isAudio - (only when setting)
  * language - (only when setting)
  * dimension - (only when setting)
  * error - (only when setting)
  * userActive - (only when setting)

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Change has been verified in an actual browser (Chome, Firefox, IE)
- [x] Unit Tests updated or fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/modal-dialog.js [hunk 1]
```diff
@@ -153,55 +153,52 @@ class ModalDialog extends Component {
    *
    * @fires ModalDialog#beforemodalopen
    * @fires ModalDialog#modalopen
-   *
-   * @return {ModalDialog}
-   *         Returns itself; method can be chained.
    */
   open() {
-    if (!this.opened_) {
```
Comment (gkatsev) [inline src/js/modal-dialog.js]: wouldn't been better to not refactor this method and just remove the chaining capabilities for this PR.

File: src/js/player.js [hunk 2]
```diff
@@ -592,7 +592,7 @@ class Player extends Component {
    *        The value to set the `Player's heigth to.
    *
    * @return {number}
-   *         The current heigth of the `Player`.
+   *         The current heigth of the `Player` when getting.
```
Comment (gkatsev) [inline src/js/player.js]: just realized there's a typo here: 'heigth' -> 'height'.

File: src/js/player.js [hunk 3]
```diff
@@ -628,14 +627,14 @@ class Player extends Component {
...
-        return this;
+        return;
       }
 
       this[privDimension] = parsedVal;
     }
 
     this.updateStyleEl_();
-    return this;
+    return;
```
Comment (gkatsev) [inline src/js/player.js]: no need for this return.

File: src/js/modal-dialog.js (hunk 1 above)
Comment (BrandonOCasey) [inline src/js/modal-dialog.js]: will revert this refactor and add to another PR

**Comment by BrandonOCasey** [comment]
@gkatsev rebased, and code review changes addressed

**Comment by BrandonOCasey** [comment]
removed method chaining changes from `Player#play()` due to #3907

**Comment by gkatsev** [comment]
Fixes https://github.com/videojs/video.js/issues/3704

**Pull request merged**